"IllegalArgumentException: requirement failed" is not actionable

I am trying to use `"scala-rewrites" % "0.1.0"` but the below error occurs.
I think the error message "requirement failed" is unclear about what is requirement, from user POV.
It would be improved if message contains "xyz is missing" or a link to document.

```
[error] scalafix.internal.v1.FileException: unexpected error processing file /home/vagrant/IdeaProjects/annofab/common/shared/src/main/scala/jp/kurusugawa/af/common/model/v1/Login.scala
[error] Caused by: java.lang.IllegalArgumentException: requirement failed: List(-Ymacro-annotations, -Xplugin:/home/vagrant/.cache/coursier/v1/https/repo1.maven.org/maven2/org/wartremover/wartremover_2.13.3/2.4.9/wartremover_2.13.3-2.4.9.jar, -Xplugin:/home/vagrant/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scalameta/semanticdb-scalac_2.13.3/4.3.18/semanticdb-scalac_2.13.3-4.3.18.jar, -Yrangepos, -P:semanticdb:synthetics:on, -P:semanticdb:targetroot:/home/vagrant/IdeaProjects/annofab/common/jvm/target/scala-2.13/meta)
[error]         at scala.Predef$.require(Predef.scala:281)
[error]         at scala.meta.internal.pc.ScalafixGlobal$.newCompiler(ScalafixGlobal.scala:36)
[error]         at fix.scala213.ExplicitNonNullaryApply.$anonfun$withConfiguration$1(ExplicitNonNullaryApply.scala:99)
[error]         at scalafix.internal.v1.LazyValue.$anonfun$computeValue$1(LazyValue.scala:24)
[error]         at scala.util.Try$.apply(Try.scala:213)
[error]         at scalafix.internal.v1.LazyValue.computeValue(LazyValue.scala:22)
[error]         at scalafix.internal.v1.LazyValue.value(LazyValue.scala:29)
[error]         at fix.scala213.ExplicitNonNullaryApply.power$lzycompute$1(ExplicitNonNullaryApply.scala:37)
[error]         at fix.scala213.ExplicitNonNullaryApply.power$1(ExplicitNonNullaryApply.scala:37)
[error]         at fix.scala213.ExplicitNonNullaryApply.$anonfun$unsafeFix$9(ExplicitNonNullaryApply.scala:50)
[error]         at fix.scala213.ExplicitNonNullaryApply.$anonfun$unsafeFix$9$adapted(ExplicitNonNullaryApply.scala:49)
[error]         at scala.Option$WithFilter.$anonfun$withFilter$1(Option.scala:332)
[error]         at scala.Option$WithFilter.$anonfun$withFilter$1$adapted(Option.scala:332)
[error]         at scala.Option.filter(Option.scala:289)
[error]         at scala.Option$WithFilter.map(Option.scala:329)
[error]         at fix.scala213.ExplicitNonNullaryApply.$anonfun$unsafeFix$8(ExplicitNonNullaryApply.scala:49)
[error]         at scala.Option$WithFilter.flatMap(Option.scala:271)
[error]         at fix.scala213.ExplicitNonNullaryApply.fix$scala213$ExplicitNonNullaryApply$$fix$1(ExplicitNonNullaryApply.scala:42)
[error]         at fix.scala213.ExplicitNonNullaryApply$$anonfun$unsafeFix$13.applyOrElse(ExplicitNonNullaryApply.scala:75)
[error]         at fix.scala213.ExplicitNonNullaryApply$$anonfun$unsafeFix$13.applyOrElse(ExplicitNonNullaryApply.scala:73)
[error]         at scala.PartialFunction$Lifted.apply(PartialFunction.scala:228)
[error]         at scala.PartialFunction$Lifted.apply(PartialFunction.scala:224)
[error]         at scala.meta.transversers.Api$XtensionCollectionLikeUI$traverser$3$.apply(Api.scala:32)
[error]         at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1(SimpleTraverser.scala:5)
[error]         at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1$adapted(SimpleTraverser.scala:5)
[error]         at scala.collection.immutable.List.foreach(List.scala:392)
[error]         at scala.meta.transversers.SimpleTraverser.apply(SimpleTraverser.scala:5)
[error]         at scala.meta.transversers.Api$XtensionCollectionLikeUI$traverser$3$.apply(Api.scala:33)
[error]         at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1(SimpleTraverser.scala:5)
[error]         at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1$adapted(SimpleTraverser.scala:5)
[error]         at scala.collection.immutable.List.foreach(List.scala:392)
[error]         at scala.meta.transversers.SimpleTraverser.apply(SimpleTraverser.scala:5)
[error]         at scala.meta.transversers.Api$XtensionCollectionLikeUI$traverser$3$.apply(Api.scala:33)
[error]         at scala.meta.transversers.Api$XtensionCollectionLikeUI.collect(Api.scala:36)
[error]         at fix.scala213.ExplicitNonNullaryApply.unsafeFix(ExplicitNonNullaryApply.scala:73)
[error]         at fix.scala213.ExplicitNonNullaryApply.fix(ExplicitNonNullaryApply.scala:26)
[error]         at scalafix.internal.v1.Rules.$anonfun$semanticPatch$1(Rules.scala:75)
[error]         at scala.collection.immutable.List.map(List.scala:286)
[error]         at scalafix.internal.v1.Rules.semanticPatch(Rules.scala:73)
[error]         at scalafix.internal.v1.MainOps$.unsafeHandleFile(MainOps.scala:232)
[error] stack trace is suppressed; run last commonJVM / Compile / scalafix for the full output
```

# My build

```scala
// plugins.sbt
addSbtPlugin("ch.epfl.scala" % "sbt-scalafix" % "0.9.17")
```

```scala
// build.sbt
scalaVersion := "2.13.3"
ThisBuild / scalafixDependencies ++= Seq(
  "org.scala-lang" %% "scala-rewrites" % "0.1.0"
)
inThisBuild(
  List(
    semanticdbEnabled := true, 
    semanticdbOptions += "-P:semanticdb:synthetics:on",
    semanticdbVersion := "4.3.18"
  )
)
```

--- a/README.md
+++ b/README.md
@@ -15,27 +15,26 @@ inThisBuild(List(
   semanticdbEnabled := true,
   semanticdbOptions += "-P:semanticdb:synthetics:on", // make sure to add this
   semanticdbVersion := scalafixSemanticdb.revision,
+  scalafixScalaBinaryVersion := CrossVersion.binaryScalaVersion(scalaVersion.value),
 ))
 ```
 
 Then run the desired rewrite(s) ([official docs][2]), in sbt:
 
 ```scala
-> scalafix dependency:fix.scala213.ExplicitNonNullaryApply@org.scala-lang:scala-rewrites:0.1.0
-> Test/scalafix dependency:fix.scala213.ExplicitNonNullaryApply@org.scala-lang:scala-rewrites:0.1.0
+> scalafixAll dependency:fix.scala213.ExplicitNonNullaryApply@org.scala-lang:scala-rewrites:0.1.1
 ```
 
 You can also add the following to your `build.sbt`:
 
 ```scala
-ThisBuild / scalafixDependencies += "org.scala-lang" %% "scala-rewrites" % "0.1.0"
+ThisBuild / scalafixDependencies += "org.scala-lang" %% "scala-rewrites" % "0.1.1"
 ```
 
 and then:
 
 ```scala
-> scalafix fix.scala213.ExplicitNonNullaryApply
-> Test/scalafix fix.scala213.ExplicitNonNullaryApply
+> scalafixAll fix.scala213.ExplicitNonNullaryApply
 ```
 
 [1]: https://scalacenter.github.io/scalafix/docs/users/installation.html

--- a/rewrites/src/main/scala/fix/scala213/ExplicitNonNullaryApply.scala
+++ b/rewrites/src/main/scala/fix/scala213/ExplicitNonNullaryApply.scala
@@ -94,10 +94,23 @@ final class ExplicitNonNullaryApply(global: LazyValue[ScalafixGlobal])
     case Type.Select(_, name: Type.Name) => name
   }
 
-  override def withConfiguration(config: Configuration) =
-    Configured.ok(new ExplicitNonNullaryApply(LazyValue.later { () =>
-      ScalafixGlobal.newCompiler(config.scalacClasspath, config.scalacOptions, Map.empty)
-    }))
+  override def withConfiguration(config: Configuration) = {
+    val compileSv = config.scalaVersion
+    val runtimeSv = scala.util.Properties.versionNumberString
+    if ((compileSv.take(4) != runtimeSv.take(4)) && config.scalacOptions.nonEmpty) {
+      Configured.error(
+        s"Scala version mismatch: " +
+        s"(1) the target sources were compiled with Scala $compileSv; " +
+        s"(2) Scalafix is running on Scala $runtimeSv. " +
+        s"To fix make scalafixScalaBinaryVersion == ${compileSv.take(4)}. " +
+        "Try `ThisBuild / scalafixScalaBinaryVersion := CrossVersion.binaryScalaVersion(scalaVersion.value)`."
+      )
+    } else {
+      Configured.ok(new ExplicitNonNullaryApply(LazyValue.later { () =>
+        ScalafixGlobal.newCompiler(config.scalacClasspath, config.scalacOptions, Map.empty)
+      }))
+    }
+  }
 
   override def afterComplete() = shutdownCompiler()
   def shutdownCompiler() = for (g <- global) nonFatalCatch { g.askShutdown(); g.close() }

